vibi-dpu/src/graph/utils.rs

```diff
@@ -25,47 +27,72 @@ pub async fn call_llm_api(prompt: String) -> Option<String> {
     let client = get_client();
     let url = "https://diff-grapher.openai.azure.com/openai/deployments/diff-grapher/chat/completions?api-version=2025-01-01-preview";
     let token = &*TOKEN;
-    let response_res = client
-        .post(url)
-        .header("api-key", format!("{}", token))
-        .header("Content-Type", "application/json")
-        .json(&json!({
-            "messages": [
-                { "role": "user", "content": prompt }
-            ],
-            "max_tokens": 4000,
-        }))
-        .send()
-        .await;
-
-    if let Err(err) = response_res {
-        log::error!("[call_llm_api] Error in calling API: {:?}", err);
-        return None;
-    }
 
-    let response = response_res.unwrap();
+    // Initial wait of 1 seconds before making the first API call
+    let initial_timeoff = 1;
+    sleep(Duration::from_secs(initial_timeoff)).await;
 
-    // Ensure we can read the response stream
-    let resp_text_res = response.text().await;
-    if let Err(err) = resp_text_res {
-        log::error!("[call_llm_api] Error reading response text: {:?}", err);
-        return None;
-    }
+    let mut wait_time = 5; // Initial backoff time in seconds
+
+    for attempt in 0..5 {
+
+        let response_res = client
+            .post(url)
+            .header("api-key", format!("{}", token))
+            .header("Content-Type", "application/json")
+            .json(&json!({
+                "messages": [
+                    { "role": "user", "content": prompt }
+                ],
+                "max_tokens": 4000,
+            }))
+            .send()
+            .await;
+
+        if let Err(err) = response_res {
+            log::error!("[call_llm_api] Error in calling API: {:?}", err);
+            continue;
```

Comment:
Does this mean that every time there is an error, it will just continue in the loop? 

If that is the case, then shouldn't the `sleep` command be here instead of at the end of the function?

Reply:
it will continue without exponential increase of wait time. Exponential increase happens only when token is exhausted. That error is not caught here, but at the end of the loop